cherryontech/unmatched
# #UM-8 Setup repo with page routing, page layout, and DaisyUI configuration

## Description

This PR adds page routing via React Router. It also adds the [DaisyUI components](https://daisyui.com/components/) library (added to the dependencies and configured in the Tailwind config file). For more effective routing, the PR refactors the folder structure by adding a views folder in which the apps pages live, alongside a Layout page which is a basic page layout that will show on every page. Currently the layout includes a page header, following the styling in the Figma designs. However, none of the buttons in the header are functional yet. Please review the comments I've left below to get more details on all of these features.

## What type of PR is this? (check all applicable)


|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓    | :sparkles: New feature     |
| ✓    | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
| ✓    | 🎨 Style                   |
|     | 🔥 Performance Improvements|
|     | :scroll: Docs              |

## Related Tickets & Documents
Closes [UM-8](https://lemon-zest.atlassian.net/jira/software/projects/UM/boards/2?selectedIssue=UM-8)

## Desktop Screenshots

![Screenshot 2023-10-25 at 10 43 21 AM](https://github.com/cherryontech/unmatched/assets/115492619/854175ec-489a-40e5-8ad5-874731b41eeb)

## What I learned
I learned how to setup React router and a more intuitive folder structure for page routing (i.e. the views folder separate from the components folder). I also learned how to create a custom color theme in DaisyUI and how the color system in DaisyUI works very well with the design system our designers have setup. 

## Testing steps

1. Switch the this branch with the following command: `git switch feature/UM-8-repo-setup`
2. Install dependencies with the following command: `npm i`
3. Run the following scripts: `npm run dev` to start Vite's dev build & `npm run tailwind` to start the Tailwind CSS build
4. View the app in browser (the `npm run dev` script will return the url you can see the app at)
5. Add "/professionals" to the end of the url to navigate to the professionals page (remove it to navigate back home). Ensure this navigation is working properly. 
6. Inspect the code to understand the routing - be sure to look closely at the `views` folder, the `Layout.jsx` file, and the `App.jsx` file
7. Inspect the code to understand the DaisyUI setup. You can see it installed in our dependencies, and then configured in the `tailwind.config.js` file. Look over DaisyUI's easy-to-read [documentation on colors}(https://daisyui.com/docs/colors/#-2) to understand how to apply the custom colors set up in `tailwind.config.js`. I've also left comments on the code in this PR to point out where DaisyUI is being used in the code.
8. Interact with the code and ask questions/make suggestions so you thoroughly understand the setup.

## What gif best describes this PR or how it makes you feel?
![hump day](https://media.giphy.com/media/uLgd9dOYWpnu5WkShY/giphy.gif?cid=ecf05e4728w6pbnz58rc6l1ycj76u3f2fmkeyops6m5o4225&ep=v1_gifs_trending&rid=giphy.gif&ct=g)







[UM-8]: https://lemon-zest.atlassian.net/browse/UM-8?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
